backends/dpdk/dpdkHelpers.cpp

```diff
@@ -1460,4 +1469,139 @@ bool ConvertStatementToDpdk::preorder(const IR::SwitchStatement *s) {
+        }
+        add_instr(new IR::DpdkMovStatement(src1Op, tmp));
+        add_instr(new IR::DpdkMovStatement(tmp, src1OpUpper));
+        if (strcmp(op, "xor") == 0) {
+            add_instr(new IR::DpdkXorStatement(tmp, tmp, src2Op));
+        } else if (strcmp(op, "or") == 0) {
+            add_instr(new IR::DpdkOrStatement(tmp, tmp, src2Op));
+        } else if (strcmp(op, "and") == 0) {
+            add_instr(new IR::DpdkAndStatement(tmp, tmp, src2Op));
+        }
+        add_instr(new IR::DpdkMovhStatement(src1Op, tmp));
+    } else {
+        auto src2OpHeaderName = src2Op->toString();
+        if (src2Op->is<IR::Member>()) {
+            src2OpHeaderName = src2Op->to<IR::Member>()->member.name;
+        }
+        src2OpHeaderName = src2OpHeaderName + "_128";
+        auto src2OpHeader = new IR::Declaration_Variable(src2OpHeaderName, Type_Header);
+        auto src2OpHeaderInstance = new IR::DpdkHeaderInstance(src2OpHeader, Type_Header);
+        structure->addHeaderInstances(src2OpHeaderInstance);
+        auto src2OpUpper =
+            new IR::Member(new IR::PathExpression("h." + src2OpHeader->name), IR::ID("upper_half"));
+        auto src2OpLower =
+            new IR::Member(new IR::PathExpression("h." + src2OpHeader->name), IR::ID("lower_half"));
+        add_instr(new IR::DpdkMovhStatement(src1OpUpper, src1Op));
+        add_instr(new IR::DpdkMovStatement(src1OpLower, src1Op));
+        add_instr(new IR::DpdkMovhStatement(src2OpUpper, src2Op));
+        add_instr(new IR::DpdkMovStatement(src2OpLower, src2Op));
+        add_instr(new IR::DpdkMovStatement(tmp, src1OpLower));
+        if (strcmp(op, "xor") == 0) {
+            add_instr(new IR::DpdkXorStatement(tmp, tmp, src2OpLower));
+        } else if (strcmp(op, "or") == 0) {
+            add_instr(new IR::DpdkOrStatement(tmp, tmp, src2OpLower));
+        } else if (strcmp(op, "and") == 0) {
+            add_instr(new IR::DpdkAndStatement(tmp, tmp, src2OpLower));
+        }
+        add_instr(new IR::DpdkMovStatement(src1Op, tmp));
+        add_instr(new IR::DpdkMovStatement(tmp, src1OpUpper));
+        if (strcmp(op, "xor") == 0) {
+            add_instr(new IR::DpdkXorStatement(tmp, tmp, src2OpUpper));
+        } else if (strcmp(op, "or") == 0) {
+            add_instr(new IR::DpdkOrStatement(tmp, tmp, src2OpUpper));
+        } else if (strcmp(op, "and") == 0) {
+            add_instr(new IR::DpdkAndStatement(tmp, tmp, src2OpUpper));
+        }
+        add_instr(new IR::DpdkMovhStatement(src1Op, tmp));
+    }
+}
+
+void ConvertStatementToDpdk::createSandboxHeader() {
+    auto fields = new IR::IndexedVector<IR::StructField>;
+    fields->push_back(new IR::StructField("upper_half", IR::Type_Bits::get(64)));
+    fields->push_back(new IR::StructField("lower_half", IR::Type_Bits::get(64)));
+    const IR::Type_Header *headerStruct = new IR::Type_Header(IR::ID("sandbox_header_t"), *fields);
+    auto name = new cstring("sandbox_header_t");
```

Comment:
can we use some name which are not valid identifier in P4 say starting with $
x=same tmp128_t

Reply:
sandbox_header$_t

tmp128$_t

Are these fine?